### ISSUE TITLE
NSProxy Test fails on Windows MSVC x86_64

### ISSUE THREAD

**Issue by hmelder** [gnustep/libs-base#278]
Test fails after commit (as seen in the CI test logs):
https://github.com/gnustep/libs-base/commit/0f9edb188bf3a05ee0d414030b4fc351757bb14a

**Comment by triplef** [comment]
```
base/NSProxy/test00.m:
Failed test:     (2022-09-27 18:44:56.871 +0000) test00.m:126 ... subclass isEqual: to proxy
Failed test:     (2022-09-27 18:44:56.871 +0000) test00.m:132 ... subclass isEqualToString: to proxy
Failed test:     (2022-09-27 18:44:56.871 +0000) test00.m:138 ... subclass compare: proxy
Failed test:     (2022-09-27 18:44:56.871 +0000) test00.m:139 ... remote compare: proxy
```

**Comment by hmelder** [comment]
I am not quite sure where this comes from. Just logging the class description returns the same description for the subclass and the proxy. The test files as well as NSProxy or NSObject were not changed between the commits.

So I have to debug it on a x86_64 Windows machine which I don't have access to rn.

**Comment by hmelder** [comment]
And the commits seem to be unrelated. The Source/NSByteCountFormatter.m commit is the only change I can imagine that are causing these issues.

**Comment by triplef** [comment]
Yeah super weird, and I can’t even reproduce it here on my machine so far...

The only thing I found comparing logs I found that between [the last run that worked](https://github.com/gnustep/libs-base/actions/runs/3061006330/jobs/4940233524) and [the first one that didn’t](https://github.com/gnustep/libs-base/actions/runs/3129127366/jobs/5077827311), libffi was updated from v3.4.2 to v3.4.3. Since I think NSProxy uses libffi via NSInvocation and there were [quite a few changes](https://github.com/libffi/libffi/compare/v3.4.2...v3.4.3) between the two libffi versions, maybe we’re exposing some sort of bug there (like a memory corruption).

I’ll keep trying to reproduce it here.

**Comment by triplef** [comment]
Might have something to do with this change: https://github.com/libffi/libffi/issues/733. While I don’t understand the specifics, the [commit causing this](https://github.com/libffi/libffi/commit/9b58d4bad688820fa9bb077c0ec5f5949ac18bd2) mentions MSVC.

The working version’s libffi configure output was:
```
checking whether byte ordering is bigendian... no
checking assembler .cfi pseudo-op support... no
checking assembler supports pc related relocs... yes
```

The broken one:
```
checking whether byte ordering is bigendian... no
grep: conftest.s: No such file or directory
checking compiler generates .cfi pseudo-ops... no
checking assembler supports pc related relocs... yes
```

**Comment by hmelder** [comment]
Thanks for looking into this. This pins the issue down to libffi which makes more sense. We could use the older version in the windows MSVC toolchain, and enable CI. We can then investigate this further.Sent from my iPhoneOn 5. Oct 2022, at 21:24, Frederik Seiffert ***@***.***> wrote:﻿
Might have something to do with this change: libffi/libffi#733. While I don’t understand the specifics, the commit causing this mentions MSVC.
The working version’s libffi configure output was:
checking whether byte ordering is bigendian... no
checking assembler .cfi pseudo-op support... no
checking assembler supports pc related relocs... yes

The broken one:
checking whether byte ordering is bigendian... no
grep: conftest.s: No such file or directory
checking compiler generates .cfi pseudo-ops... no
checking assembler supports pc related relocs... yes

**Comment by triplef** [comment]
Ok I was able to reproduce it after properly updating libffi to v3.4.3, and unfortunately also on libffi master (so the reversion of the above-linked commit didn’t fix it).

Unfortunately I don’t have the time to investigate this further atm, so for the time being we’re now using v3.4.2 (https://github.com/gnustep/tools-windows-msvc/commit/551015a9ec1d49824c44c565db042d2bc0ec7e0d) and I again disallowed test failures on Windows (https://github.com/gnustep/libs-base/commit/f1961f47e9734a0b94ebdecd590f7d7f60e7d3bf).

**Comment by hmelder** [comment]
Maybe running the libffi testsuite on Windows provides more in-depth results.Sent from my iPhoneOn 5. Oct 2022, at 21:24, Frederik Seiffert ***@***.***> wrote:﻿
Might have something to do with this change: libffi/libffi#733. While I don’t understand the specifics, the commit causing this mentions MSVC.
The working version’s libffi configure output was:
checking whether byte ordering is bigendian... no
checking assembler .cfi pseudo-op support... no
checking assembler supports pc related relocs... yes

The broken one:
checking whether byte ordering is bigendian... no
grep: conftest.s: No such file or directory
checking compiler generates .cfi pseudo-ops... no
checking assembler supports pc related relocs... yes
